core/src/main/java/com/netflix/iceberg/TableOperations.java

```diff
@@ -56,27 +55,17 @@
   void commit(TableMetadata base, TableMetadata metadata);
 
   /**
-   * Create a new {@link InputFile} for a path.
-   *
-   * @param path a string file path
-   * @return an InputFile instance for the path
+   * @return a {@link com.netflix.iceberg.FileIO} to read and write table data and metadata files
    */
-  InputFile newInputFile(String path);
+  FileIO io();
 
   /**
-   * Create a new {@link OutputFile} in the table's metadata store.
-   *
-   * @param filename a string file name, not a full path
-   * @return an OutputFile instance for the path
-   */
-  OutputFile newMetadataFile(String filename);
-
-  /**
-   * Delete a file.
-   *
-   * @param path path to the file
+   * Given the name of a metadata file, return the full path that points to that data file.
```

Comment:
I think it is slightly different. Given a filename, create a full path using an appropriate location for metadata files.